desc/objectives/objective_funs.py

```diff
@@ -63,62 +70,9 @@ def _set_derivatives(self):
                 self._deriv_mode = "batched"
```

Comment:
With these deletions, do we still need this function? Maybe just move the code to init. If not, change the documentation since it doesn't set derivatives but sets the method to calculate derivatives.

Reply:
I think we don't want to put this in init because it needs to be called after the sub-objectives get built, otherwise some of the logic for picking forward or reverse mode based on size of outputs etc won't work correctly.

Reply:
Ok I am fine with new documentation